tap_universal_file/streams.py

```diff
@@ -478,3 +479,234 @@ def _get_readers(
+
+        Yields:
+            A dictionary or string representing a field.
+        """
+        strategy = self.config["parquet_type_coercion_strategy"]
+        if strategy == "convert":
+            for reader, _, _ in self._get_readers():
+                yield from (
+                    {"name": name, "type": reader_type}
+                    for name, reader_type in zip(
+                        reader.schema.names,
+                        reader.schema.types,
+                    )
+                )
+            return
+        if strategy == "envelope":
+            yield "record"
+            return
+        msg = f"The coercion strategy '{strategy}' is not valid."
+        raise ValueError(msg)
+
+    def _get_property(self, field: dict | str) -> dict[str, dict[str, list[str]]]:
+        """Converts a field into a JSON schema fragment based on coercion strategy.
+
+        Args:
+            field: The name of the field to get a property from.
+
+        Raises:
+            ValueError: If the coercion strategy provided is invalid.
+
+
+        Returns:
+            A dictionary containing a representation of the field as a property.
+        """
+        strategy = self.config["parquet_type_coercion_strategy"]
+        if strategy == "convert":
+            field_type, field_format = self._type_convert(field["type"])
+            type_dict = {"type": [field_type]}
+            if field_format is not None:
+                type_dict.update({"format": field_format})
+            return {field["name"]: type_dict}
+        if strategy == "envelope":
+            return {field: {"type": ["null", "object"]}}
+        msg = f"The coercion strategy '{strategy}' is not valid."
+        raise ValueError(msg)
+
+    def _type_convert(  # noqa: C901 PLR0912
+        self,
+        field_type: pa.DataType,
+    ) -> tuple[str, str | None]:
+        """Attempt to coerce a Parquet schema type to a JSON schema type.
+
+        Args:
+            field_type: The field to be converted.
+
+        Raises:
+            NotImplementedError: If the field type passed in is not implemented.
+
+        Returns:
+            A JSON schema representation of field_type.
+        """
+        type_tuple = None
+        if pa.types.is_null(field_type):
+            type_tuple = ("null", None)
+        if pa.types.is_boolean(field_type):
+            type_tuple = ("boolean", None)
+        if pa.types.is_integer(field_type):
+            type_tuple = ("integer", None)
+        if pa.types.is_floating(field_type):
+            type_tuple = ("number", None)
+        if pa.types.is_time(field_type):
+            type_tuple = ("string", "time")
+        if pa.types.is_date(field_type):
+            type_tuple = ("string", "date")
+        if pa.types.is_timestamp(field_type):
+            type_tuple = ("string", "date-time")
+        if pa.types.is_duration(field_type):
+            type_tuple = ("string", "duration")
```

Comment:
duration isn't a part of json schema 4, I"d go with string here instead

Reply:
Fixed.